py/torch_tensorrt/dynamo/_compiler.py

```diff
@@ -452,6 +458,33 @@ def contains_metadata(gm: torch.fx.GraphModule) -> bool:
             )
             continue
 
+        if name not in submodule_node_dict:
+            raise ValueError(
+                f"node_name: {name} does not exist in the submodule node dictionary"
+            )
+
+        # set the submodule metadata back to the parent trt_module_node
+        metadata_list = get_output_metadata(submodule)
+        assert len(metadata_list) > 0
+        if "val" not in submodule_node_dict[name].meta:
+            meta_val_list = [
+                metadata["val"] for metadata in metadata_list if "val" in metadata
+            ]
+            submodule_node_dict[name].meta["val"] = meta_val_list
+            logger.debug(
+                f"Updated val metadata for node: {name} with its corresponding submodule outputs"
+            )
+        if "tensor_meta" not in submodule_node_dict[name].meta:
+            tensor_meta_list = [
+                metadata["tensor_meta"]
+                for metadata in metadata_list
+                if "tensor_meta" in metadata
+            ]
+            submodule_node_dict[name].meta["tensor_meta"] = tensor_meta_list
+            logger.debug(
+                f"Updated tensor_meta metadata for node: {name} with its corresponding submodule outputs"
+            )
```

Comment:
Consider merging these in a single update to avoid code duplication. Eg: 
```py
metadata_keys = ["val", "tensor_meta"]
for key in metadata_keys:
     if key not in submodule_node_dict[name].meta:
          .......
          logger.debug(f"Updated {key} metadata for node ........")
          break # if val is found no need to check tensor_meta
```